plugins/woocommerce/src/Blocks/BlockTypes/ProductResultsCount.php

```diff
@@ -35,19 +35,34 @@ protected function get_block_type_script( $key = null ) {
 	 * @return string Rendered block output.
 	 */
 	protected function render( $attributes, $content, $block ) {
+		// Buffer the result count and use it as the block's frontend content.
 		ob_start();
+		echo '<div>';
 		woocommerce_result_count();
+		echo '</div>';
 		$product_results_count = ob_get_clean();
 
-		$classes_and_styles = StyleAttributesUtils::get_classes_and_styles_by_attributes( $attributes );
-		$classname          = isset( $attributes['className'] ) ? $attributes['className'] : '';
+		$p = new \WP_HTML_Tag_Processor( $product_results_count );
 
-		return sprintf(
-			'<div class="woocommerce wc-block-product-results-count wp-block-woocommerce-product-results-count %1$s %2$s" style="%3$s">%4$s</div>',
```

Comment:
@ObliviousHarmony I'm curious why we don't add the navigation ID here directly but use `WP_HTML_Tag_Processor` 🤔?

Reply:
From the diff, we don't have to modify any rendered content. We add the directive to the wrapper div, which is created within this block.

Reply:
At this point `sprintf()` would have five placeholders and added classes, a specific class name, a style attribute, a navigation attribute, and the HTML content. This felt like it's getting a bit excessive and so I wanted to refactor it rather than adding to it.

`WP_HTML_Tag_Processor` _seemed_ like a good idea since all I was doing was adding attributes to a block wrapper. Is the best practice to just keep expanding `sprintf()`? It felt like there should already be some kind of block wrapper element function but I couldn't find one.

Reply:
> At this point sprintf() would have five placeholders and added classes

I agree that we shouldn't use `sprintf` now because of this reason.

> It felt like there should already be some kind of block wrapper element function but I couldn't find one.It felt like there should already be some kind of block wrapper element function but I couldn't find one.

What about passing additional attributes to `get_block_wrapper_attributes` and then [printing](https://github.com/WordPress/block-development-examples/blob/trunk/plugins/block-dynamic-rendering-64756b/src/render.php) them all at once?

Reply:
That's a neat function @dinhtungdu. I wonder though, how well does this play with `StyleAttributesUtils::get_classes_and_styles_by_attributes()`? We seem to be performing our own parsing of attributes and I wonder if there's any overlap between the two that might cause issues? It feels like the result of our static utility could just be passed to `get_block_wrapper_attributes()` (with an alias to use `class` and `style` instead of the current `classes` and `styles` indices)

Reply:
> It feels like the result of our static utility could just be passed to get_block_wrapper_attributes() (with an alias to use class and style instead of the current classes and styles indices)

Yeah, that's what I thought too. That function merges extra `class`, `id`, and `style` attributes (if passed) with ones from block supports. I don't think there is any issue with that approach. Besides, if we use `get_block_wrapper_attributes()`, we don't have to manually check and get the value of `className` from `$attributes` like we're doing. Not to mention `id`, I believe that, currently, the block won't display the ID if set in the editor.

Reply:
Do you think this is something that needs to be tackled right now or is this more of a discussion about technical debt?

Reply:
> Not to mention id, I believe that, currently, the block won't display the ID if set in the editor.

This is not true because the block doesn't support `anchor`.

> Do you think this is something that needs to be tackled right now or is this more of a discussion about technical debt?

Given that the block is still working as expected and doesn't introduce any bugs, we don't need to update this right now. But it's good to discuss the wrapper to find an optimal way to handle the attribute and then update our blocks.